custom_components/waste_collection_schedule/waste_collection_schedule/source/korneuburg_stadtservice_at.py

```diff
@@ -0,0 +1,181 @@
+import json
+from urllib.parse import urljoin
+
+import requests
+from bs4 import BeautifulSoup
+from requests.adapters import HTTPAdapter, Retry
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+from waste_collection_schedule.service.ICS import ICS  # type: ignore[attr-defined]
+
+TITLE = 'Müllabfuhr Korneuburg'
+DESCRIPTION = 'Source for Stadtservice Korneuburg'
+URL = 'https://www.korneuburg.gv.at'
+
+# Mapping of teilgebiete to calendar urls
+WASTE_TYPE_URLS = {
+    '1': ('Biomuell_3', 'Restmuell_3', 'Papier_2', 'Gelber_Sack_4'),
+    '2': ('Biomuell_4', 'Restmuell_2', 'Papier_3', 'Gelber_Sack_1'),
+    '3': ('Biomuell_1', 'Restmuell_1', 'Papier_1', 'Gelber_Sack_2'),
+    '4': ('Biomuell_2', 'Restmuell',   'Papier',   'Gelber_Sack_3')
+}
+
+TEST_CASES = {
+    "Rathaus": {"street_name": "Hauptplatz", "street_number": 39},                                         # Teilgebiet 4
+    "Rathaus using Teilgebiet": {"street_name": "Some Street", "street_number": "1A", "teilgebiet": "4"},  # Teilgebiet 4
+    "Werft": {"street_name": "Am Hafen", "street_number": 6}                                               # Teilgebiet 2
+}
+
+
+def http_get_retry(params, n_retries=5):
+    """HTTP GET with n retries """
+    s = requests.Session()
+
+    retries = Retry(total=n_retries,
```

Comment:
Just to be curious: Is their service so unreliable that a special retry handling is required? I have never seen a need for this in any other source.

Reply:
When I was writing the code I often got 502 responses and this helped so I could run all the tests. 

Today the tests ran quite fast so I think it was a temporary issue. If you prefer I can remove this function.

Reply:
Yes, I would prefer to start without this special code. Please remove it. Thanks.

Reply:
Okay. Done